blog/articles/orchestrating-cloud-events-with.md

```diff
@@ -0,0 +1,187 @@
+
+## Orchestrate
+
+The **Payment Reminder** and **Reservation Time Out** states, shown in the model above, represent when we cross the line from simply listening to events to actually emitting Cloud Events from a workflow model. Now the workflow model is not only helping you to visualize and understand what your Cloud-Native applications are doing but driving and interacting with your services.
+
+While using a workflow engine to orchestrate Cloud Events, you have a bunch of new tools at your fingertips to deal with more complex scenarios.
+
+Version 3 of the Workflow Model shows a more complex diagram, where a subprocess can be used to group a set of states together. By doing so, you can quickly identify stages inside your workflows, as well as register events for these stages such as timers or message events.
+
+![Workflow Model V3](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v3.png?raw=true)
+
+The worfklow can react on the Cloud Event **Customer Abandoned Queue** at any given time inside the **Customer Buying Tickets** stage. This will trigger the Customer **Clean Up** event to garbage collect all the data related with the Customer Session from all the services caching data.
+
+In the following screenshot you can see how inside Camunda Operate, you can quickly visualize how many instances of the workflows are at a given stage as well as how many workflows have finished, and in which state they finished:
+
+![Workflow Model V3 in Operate](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v3-operate-stage.png?raw=true)
+
+Another advantage of using a workflow engine is flow control. By using Flow Control elements such as Exclusive Gateways, you can delegate some high-level decisions (usually encoding business logic) to the workflow engine:
+
+![Workfloe Model V4](https://github.com/salaboy/orchestrating-cloud-events/blob/master/imgs/tickets-v4.png?raw=true)
+
+As you can see in the above model, the exclusive gateway is being used to choose between two different paths based on a condition. In this case, the condition is evaluating how many tickets are being reserved by the customer and, based on that, the model is choosing between a normal credit card payment or a more complex money transfer.
+
+## Architecture & Next Steps
+
+From an architectural perspective, the integration between the Workflow Engine and Knative is quite simple. It relies on **Cloud Events** and a component that is in charge to route the events to the right workflows, and the other way around.
+
+The [Zeebe Cloud Events Router](https://github.com/zeebe-io/zeebe-cloud-events-router) is in charge of exposing an endpoint where events can be forwarded and, at the same time, understand where to push events if workflow models are emitting events.
```

Comment:
I suppose you mean "push events **when** workflow models ...", not if.

Reply:
@lionelvillard ok.. I've rephrased it.. thanks.. I think that it is much clear now.